utils/include/CJSONFormatSaver.cpp

```diff
@@ -4,31 +4,67 @@
 #include "CJSONFormatter.h"
 #include "CJSONFormatSaver.h"
 
+CJSONFormatSaver::CJSONFormatSaver(const std::string& path_to_file) :
+	m_path_to_file(path_to_file), m_number_of_spaces(3)
+{ };
+
 bool CJSONFormatSaver::TrySaveToFile(CJSONFormatter& formatted_data)
 {
-	std::ifstream json_file_to_read;
-	json_file_to_read.open(m_path_to_file);
-	nlohmann::json buff;
-	if (Utils::IsFileEmpty(json_file_to_read))
+	if (Utils::TryCreateFileIfNotExist(m_path_to_file))
+	{
+		//write to log??
+	}
+
+	std::ofstream JSON_file_to_save(m_path_to_file, std::ios::binary || 
+									std::ios::app);
+
+	if (!JSON_file_to_save.is_open())
 	{
-		buff = *formatted_data.GetJSONFormattedData();
+		JSON_file_to_save.close();
```

Comment:
Does ofsteam call close() function in the destructor? If yes then we do not need to call close() before return in another case it will be good to have a wrapper which closes a file in the destructor

Reply:
ofstream calls close() in the destructor, but, also, this function returns bool, so we can use std::ios::fail() after ofstream.close() to check does the file closed properly and return result of this operation from function.

Reply:
I removed .close()